examples/test_xworld3d.cpp

```diff
@@ -18,25 +18,64 @@
 using namespace simulator::xworld3d;
 using namespace simulator;
 
+DECLARE_bool(task_groups_exclusive);
+
+void game_reset_with_teacher(SimulatorPtr game, TeacherPtr teacher) {
+    game->reset_game();
+    teacher->reset_after_game_reset();
+    teacher->teach();
+}
+
```

Comment:
why having this function? We should also wrap xworld3d with simulator_interface and call its reset_game

Reply:
I'll add xworld3d in demo_interface.cpp

Reply:
?